memory/log.txt

```diff
@@ -1,76 +1,103 @@
-21      1048576                  346880              186                 
-22      2097152                  508630              300                 
-23      4194304                  1477771             6869                
-24      8388608                  4135254             734                 
-25      16777216                 5359725             968                 
-26      33554432                 15250               13000               
-27      67108864                 5334                4553                
-28      134217728                4238                4544                
-29      268435456                4176                5120                
-30      536870912                4059                6818                
-31      1073741824               5032                10706               
-32      2147483648               4465                10362               
-33      4294967296               4402                10506               
+0       1                        473                 163                 
+1       2                        200                 115                 
+2       4                        164                 105                 
+3       8                        122                 80                  
+4       16                       170                 78                  
+5       32                       172                 87                  
+6       64                       200                 100                 
+7       128                      385                 70                  
+8       256                      248                 82                  
+9       512                      317                 102                 
+10      1024                     381                 96                  
+11      2048                     343                 341                 
+12      4096                     785                 112                 
+13      8192                     4993                152                 
+14      16384                    3966                112                 
+15      32768                    11941               136                 
+16      65536                    26086               204                 
+17      131072                   69981               187                 
+18      262144                   150670              423                 
+19      524288                   53354               1525                
+20      1048576                  866863              2804                
+21      2097152                  899034              483                 
+22      4194304                  1189385             1004                
+23      8388608                  2389343             839                 
+24      16777216                 5158120             1421                
+25      33554432                 26193               17083               
+26      67108864                 10045               9273                
```

Comment:
Can You explain why allocation of  2**25 bytes faster than allocation of 2**24 bytes?

Reply:
I think maybe in linux calloc function uses simple allocation and memset in cases, where we need lesser than certain amount bytes. So, in another cases it switches to more efficient way to get zero-initialized memory. For example - free pages (2^24 bytes - 4096 pages, 1 page - 4096 bytes). I'd glad to receive your opinion about this result